src/core_modules/capture-core-utils/RulesEngine/executionService/executionService.js

```diff
@@ -646,15 +645,10 @@ export default function getExecutionService(variableService, dateUtils) {
             })
             .filter(ruleEffectsForRule => ruleEffectsForRule)
             .reduce((accRuleEffects, effectsForRule) => [...accRuleEffects, ...effectsForRule], []);
-    };
 
-    return {
-        executeRules(programRulesContainer, executingEvent, evs, allDataElements, allTrackedEntityAttributes, selectedEntity, selectedEnrollment, selectedOrgUnit, optionSets, flags) {
-            if (!programRulesContainer.programRules) {
-                return null;
-            }
-            return internalExecuteRules(programRulesContainer, executingEvent, evs, allDataElements, allTrackedEntityAttributes, selectedEntity, selectedEnrollment, selectedOrgUnit, optionSets, flags);
-        },
-        convertDataToBaseOutputValue: convertRuleEffectDataToOutputBaseValue,
+        const processRulesEffects = getRulesEffectsProcessor(convertRuleEffectDataToOutputBaseValue, rulesEffectsValueConverter);
```

Comment:
I am also realising that moving this inside the execution service is wrong since effects dont have anything to do with executionService really.

Reply:
I agree, I would prefer to keep the effects processing separate.